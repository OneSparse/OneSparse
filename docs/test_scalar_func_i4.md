# Scalar 'integer' Functions

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
```
set client_min_messages = 'WARNING';
create extension if not exists onesparse;

```
Test construction of min, zero and max values:
```
select '-1'::integer::scalar;
select '0'::integer::scalar;
select '1'::integer::scalar;

```
Test setting a scalar value from max to 2
```
select set('1'::integer::scalar, 2);

```
Test various casting functions used by the CREATE CAST machinery:
```
select scalar_integer((-1)::integer);
select scalar_integer((0)::integer);
select scalar_integer((1)::integer);
select integer_scalar((-1)::integer::scalar);
select integer_scalar((0)::integer::scalar);
select integer_scalar((1)::integer::scalar);
```