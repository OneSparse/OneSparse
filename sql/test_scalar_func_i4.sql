set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select '-1'::integer::scalar;
select '0'::integer::scalar;
select '1'::integer::scalar;
