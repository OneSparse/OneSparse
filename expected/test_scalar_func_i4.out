-- # Scalar 'integer' Functions
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
-- Test construction of min, zero and max values:
select '-1'::integer::scalar;
 scalar 
--------
 i4:-1
(1 row)

select '0'::integer::scalar;
 scalar 
--------
 i4:0
(1 row)

select '1'::integer::scalar;
 scalar 
--------
 i4:1
(1 row)

-- Test setting a scalar value from max to 2
select set('1'::integer::scalar, 2);
 set  
------
 i4:2
(1 row)

-- Test various casting functions used by the CREATE CAST machinery:
select scalar_integer((-1)::integer);
 scalar_integer 
----------------
 i4:-1
(1 row)

select scalar_integer((0)::integer);
 scalar_integer 
----------------
 i4:0
(1 row)

select scalar_integer((1)::integer);
 scalar_integer 
----------------
 i4:1
(1 row)

select integer_scalar((-1)::integer::scalar);
 integer_scalar 
----------------
             -1
(1 row)

select integer_scalar((0)::integer::scalar);
 integer_scalar 
----------------
              0
(1 row)

select integer_scalar((1)::integer::scalar);
 integer_scalar 
----------------
              1
(1 row)

